dependency/actions/compile/entrypoint/main.go

```diff
@@ -0,0 +1,480 @@
+package main
+
+import (
+  _ "embed"
+  "encoding/json"
+  "errors"
+  "flag"
+  "fmt"
+  "github.com/paketo-buildpacks/packit/v2/fs"
+  "log"
+  "os"
+  "os/exec"
+  "path/filepath"
+  "golang.org/x/exp/slices"
+)
+
+type ExtensionsData struct {
+  MD5           string
+  Name          string
+  Version       string
+}
+
+type CommandWithArgs struct {
+  Command string
+  Args []string
+}
+
+func main() {
+  var (
+    version   string
+    outputDir string
+    target    string
+  )
+
+  flag.StringVar(&version, "version", "", "PHP dependency version")
+  flag.StringVar(&outputDir, "output-dir", "", "Output directory for PHP dependency artifact")
+  flag.StringVar(&target, "target", "", "Dependency target")
+  flag.Parse()
+
+  if version == "" {
+	  fail(errors.New(`missing required input "version"`))
+  }
+
+  if outputDir == "" {
+	  fail(errors.New(`missing required input "output-dir"`))
+  }
+
+  if target == "" {
+	  fail(errors.New(`missing required input "target"`))
+  }
+
+  dependencyDir, err := os.MkdirTemp("", "php-dist")
+  if err != nil {
+	  log.Fatal(err)
+  }
+
+  err = installPHP(version, dependencyDir)
+  if err != nil {
+	  log.Fatal(err)
+  }
+
+  err = installExtensions(dependencyDir)
+  if err != nil {
+	  log.Fatal(err)
+  }
+
+  err = copyFiles(dependencyDir)
+  if err != nil {
+	  log.Fatal(err)
+  }
+
+  err = createArtifact(version, target, outputDir, dependencyDir)
+  if err != nil {
+	  log.Fatal(err)
+  }
+}
+
+func copyFiles(dependencyDir string) error {
```

Comment:
To the functions, can you add a comment for each just giving a description of what its for?

Reply:
@sophiewigmore I'll try to give this function a more descriptive name. I think adding a ton of comments to explain what a function does is sort of a tell that the function name is not useful. Would that be okay?

Reply:
this request isn't specific to this function. In general the compilation code is pretty complicated and if ever have to change things, it would help be helpful to have a bit of context on why things are the way they are